### PR TITLE
Add tests for `Consensus restart after more than f nodes crash`

### PR DESCRIPTION
**Pull request by JekaMas**
15 skipped test cases and a bit refactoring

### CODE REVIEW

File: consensus/test/tendermint_test.go [hunk 1]
```diff
@@ -598,3 +783,40 @@ func runHook(validatorHook hook, test *testCase, block *types.Block, validator *
...
+
+func hookStopNode(nodeIndex int, blockNum uint64) hook {
+	return func(block *types.Block, validator *testNode, tCase *testCase, currentTime time.Time) error {
+		if block.Number().Uint64() == blockNum {
+			err := validator.stopNode()
+			if err != nil {
+				return err
+			}
+
+			tCase.mu.Lock()
```
Comment (b00ris) [inline consensus/test/tendermint_test.go]: It'd be better if you hide mutex manipulations inside setter.

File: consensus/test/tendermint_test.go [hunk 2]
```diff
@@ -598,3 +783,40 @@ func runHook(validatorHook hook, test *testCase, block *types.Block, validator *
...
+			tCase.mu.Unlock()
+		}
+
+		return nil
+	}
+}
+
+func hookStartNode(nodeIndex int, durationAfterStop float64) hook {
+	return func(block *types.Block, validator *testNode, tCase *testCase, currentTime time.Time) error {
+		tCase.mu.RLock()
```
Comment (b00ris) [inline consensus/test/tendermint_test.go]: Same

**Comment by yazzaoui** [comment]
lgtm, are the skipped test solved by https://github.com/clearmatics/autonity/pull/212 ?

**Comment by JekaMas** [comment]
@yazzaoui I've run this test only on the `master` branch. They all are failing on it. So then we can merge the PR into the master and then check if any solution works.

**Comment by aneequesafdar** [comment]
I guess you will remove the skip function from the test cases once we are ready to test with basic wal or something else right?

**Pull request merged**